contracts/LotteryNFT.sol

```diff
@@ -1,59 +1,236 @@
+        //     ticketsPage = userTickets_[_user][_lotteryId][start:end];
+        //     return ticketsPage;
+        // }
+        return userTickets_[_user][_lotteryId];
     }
-    function burn(uint256 tokenId) external onlyOwner {
-        _burn(tokenId);
+
+    //-------------------------------------------------------------------------
+    // STATE MODIFYING FUNCTIONS 
+    //-------------------------------------------------------------------------
+
+    /**
+     * @param   _to The address being minted to
+     * @param   _numberOfTickets The number of NFT's to mint
+     * @notice  Only the lotto contract is able to mint tokens. 
+        // uint8[][] calldata _lottoNumbers
+     */
+    function batchMint(
+        address _to,
+        uint256 _lottoId,
+        uint8 _numberOfTickets,
+        uint16[] calldata _numbers,
+        uint8 sizeOfLottery
+    )
+        external
+        onlyLotto()
+        returns(uint256[] memory)
+    {
+        // Storage for the amount of tokens to mint (always 1)
+        uint256[] memory amounts = new uint256[](_numberOfTickets);
+        // Storage for the token IDs
+        uint256[] memory tokenIds = new uint256[](_numberOfTickets);
+        for (uint8 i = 0; i < _numberOfTickets; i++) {
+            // Incrementing the tokenId counter
+            tokenIdsCount_ = tokenIdsCount_.add(1);
+            tokenIds[i] = tokenIdsCount_;
+            amounts[i] = 1;
+            // Getting the start and end position of numbers for this ticket
+            uint16 start = uint16(i.mul(sizeOfLottery));
+            uint16 end = uint16((i.add(1)).mul(sizeOfLottery));
+            // Splitting out the chosen numbers
+            uint16[] calldata numbers = _numbers[start:end];
+            // Storing the ticket information 
+            ticketInfo_[tokenIdsCount_] = TicketInfo(
+                _to,
+                numbers,
+                false,
+                _lottoId
+            );
+            userTickets_[_to][_lottoId].push(tokenIdsCount_);
+        }
+        // Minting the batch of tokens
+        _mintBatch(
+            _to,
+            tokenIds,
+            amounts,
+            msg.data
+        );
+        // Emitting relevant info
+        emit InfoBatchMint(
+            _to, 
+            _lottoId,
+            _numberOfTickets, 
+            tokenIds
+        ); 
+        // Returns the token IDs of minted tokens
+        return tokenIds;
     }
-    function getClaimStatus(uint256 tokenId) external view returns (bool) {
-        return claimInfo[tokenId];
+
+    function claimTicket(uint256 _ticketID, uint256 _lotteryId) external onlyLotto() returns(bool) {
+        require(
+            ticketInfo_[_ticketID].claimed == false,
+            "Ticket already claimed"
+        );
+        require(
+            ticketInfo_[_ticketID].lotteryId == _lotteryId,
+            "Ticket not for this lottery"
+        );
+        uint256 maxRange = ILottery(lotteryContract_).getMaxRange();
```

Comment:
it may cause problem if the lotteryId's range changes. It should be read from a function that includes the `lotteryId`